src/osd/PG.cc

```diff
@@ -3298,7 +3298,11 @@ void PG::reg_next_scrub()
       (info.stats.stats_invalid && g_conf->osd_scrub_invalid_stats)) {
     scrubber.scrub_reg_stamp = utime_t();
   } else {
-    scrubber.scrub_reg_stamp = info.history.last_scrub_stamp;
+    if (info.history.last_scrub_stamp + cct->_conf->osd_scrub_min_interval <
+        info.history.last_deep_scrub_stamp + cct->_conf->osd_deep_scrub_interval)
+      scrubber.scrub_reg_stamp = info.history.last_scrub_stamp;
+    else
+      scrubber.scrub_reg_stamp = info.history.last_deep_scrub_stamp;
```

Comment:
I don't think this really affects anything, because of a quirk in the way OSD::sched_scrub() works.  It doesn't even consider deep scrubbing until at least a regular scrub would be getting initiated.  OSD::sched_scrub() only considers load and whether a regular scrub should happen.  PG::sched_scrub() may promote that to a deep scrub if that is appropriate.
 
Also, last_deep_scrub_stamp is always <= last_scrub_stamp because when a deep scrub completes it updates both last_scrub_stamp and last_deep_scrub_stamp.  You could describe last_scrub_stamp as the last time ANY scrub happened.